jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java

```diff
@@ -0,0 +1,394 @@
+        Map<String, Node> result = new HashMap<>();
+
+        if(uri != null) {
+            response = client.prepareGet(indexName, docDef.getEntityField(), uri).get();
+            if(response != null && !response.isSourceEmpty()) {
+                String entityField = response.getId();
+                Node entity = NodeFactory.createURI(entityField) ;
+                result.put(docDef.getEntityField(), entity);
+                Map<String, Object> source = response.getSource();
+                for (String field: docDef.fields()) {
+                    Object fieldResponse = source.get(field);
+
+                    if(fieldResponse == null) {
+                        //We wont return it.
+                        continue;
+                    }
+                    else if(fieldResponse instanceof List<?>) {
+                        //We are storing the values of fields as a List always.
+                        //If there are values stored in the list, then we return the first value,
+                        // else we do not include the field in the returned Map of Field -> Node Mapping
+                        List<?> responseList = (List<?>)fieldResponse;
+                        if(responseList != null && responseList.size() > 0) {
+                            String fieldValue = (String)responseList.get(0);
+                            Node fieldNode = NodeFactoryExtra.createLiteralNode(fieldValue, null, null);
+                            result.put(field, fieldNode);
+                        }
+                    }
+                }
+            }
+        }
+
+        return result;
+    }
+
+    @Override
+    public List<TextHit> query(Node property, String qs) {
+
+        return query(property, qs, MAX_RESULTS);
+    }
+
+    /**
+     * Query the ElasticSearch for the given Node, with the given query String and limit.
+     * @param property the node property to make a search for
+     * @param qs the query string
+     * @param limit limit on the number of records to return
+     * @return List of {@link TextHit}s containing the documents that have been found
+     */
+    @Override
+    public List<TextHit> query(Node property, String qs, int limit) {
+
+        qs = parse(qs);
+        LOGGER.debug("Querying ElasticSearch for QueryString: " + qs);
+        SearchResponse response = client.prepareSearch(indexName)
+                .setTypes(docDef.getEntityField())
+                .setQuery(QueryBuilders.queryStringQuery(qs))
+                .setFrom(0).setSize(limit)
+                .get();
+
+        List<TextHit> results = new ArrayList<>() ;
+        for (SearchHit hit : response.getHits()) {
+
+            Node literal;
+            String field = (property != null) ? docDef.getField(property) : docDef.getPrimaryField();
+            List<String> value = (List<String>)hit.getSource().get(field);
+            if(value != null) {
+                literal = NodeFactory.createLiteral(value.get(0));
```

Comment:
This is related to the jena-text functionality for [storing literal values](https://jena.apache.org/documentation/query/text-query.html#storing-literal-values). If you use a query like `(?s ?score ?literal) text:query 'word'`, then `?s` will be bound to the entity/subject, `?score` to the score returned by the index, and `?literal` to the original literal value. For the Lucene implementation, storing of literal valued has to be explicitly enabled for it to work, but for the ES backend, you need to keep the original values anyway because of the document-per-entity approach. My thought here was that since the ES index should know both the original value and the language tag, it should be possible to return a language-tagged value as the literal. 

This would be useful for some applications; e.g. my application Skosmos relies on this feature (including the original language tags) to avoid extra TDB lookups and thus can use simpler SPARQL queries that also perform better than before this feature was implemented in jena-text.